### PR TITLE
Modify autoplay default by taking into account prefers-reduced-motion

### PR DESCRIPTION
**Pull request by dannycalleri**
👇👇👇
> **WARNING:** the `autoplay` property should be actually renamed to something like `preferredAutoplay` because it can be ignored depending on the value of the `prefers-reduced-motion` media query. I decided to avoid this change now, because it will break existing clients. We can do it on a separate PR

☝️ ☝️ ☝️ 

This PR introduces a default value for the `autoplay` prop, by taking into account the `prefers-reduced-motion` [media query](https://web.dev/prefers-reduced-motion/).

If the host application doesn't specify `@autoplay={{true}}`, the component checks the user preference about motion, if no preference has been given, the component will autoplay the animation; otherwise, if the user prefers reduced motion, the animation won't play. It will follow a similar logic when `@autoplay={{true}}`.

When `@autoplay={{false}}`, the component will never play the animation automatically.

I'm also modifying the visibility of a few class methods/properties to improve the interface.

**Details**: https://web.dev/prefers-reduced-motion/

### CODE REVIEW

File: addon/components/lottie.ts [hunk 1]
```diff
@@ -76,13 +77,35 @@ export default class LottieComponent extends Component<LottieArgs> {
...
+      'change',
+      this.handleReducedMotionPreferenceChange
+    );
+
     if (this.animation) {
       this.animation.destroy();
     }
   }
+
+  private handleReducedMotionPreferenceChange = () => {
```
Comment (mrloop) [inline addon/components/lottie.ts:100]: sand: this is maybe just my preference so please resolve as you like but you could avoid an anonymous function and bind to this doing the following

```js
@action private handleReducedMotionPreferenceChange() {
```

**Comment by mrloop** [comment]
Can you test for a lottie animation playing? It'd be good to have a test for the default value and for how `prefers-reduced-motion` affects playback.

**Comment by dannycalleri** [comment]
> Can you test for a lottie animation playing? It'd be good to have a test for the default value and for how `prefers-reduced-motion` affects playback.

Let me try, I think I will need to write a mock and maybe refactor the code a little bit 👍

File: addon/components/lottie.ts (hunk 1 above)
Comment (dannycalleri) [reply addon/components/lottie.ts]: I did the change, it looks more "ember style".
Now that I'm thinking about it, this is not a real "action" BTW; I believe the `@action` decorator is more for callbacks triggered by user actions, isn't it? But anyway it doesn't make a real difference, I like it more like you suggested 👍

**Comment by dannycalleri** [comment]
@mrloop I tried to test this as much as I could taking into account your suggestions, WDYT?

File: addon/components/lottie.ts (hunk 1 above)
Comment (mrloop) [reply addon/components/lottie.ts]: Yeah mostly used for binding function to component for calling from templates. Not sure what other peoples thought on using it like this 😅

Comment (dannycalleri) [reply addon/components/lottie.ts]: Seems a fair way to use it anyway 👍

**Pull request merged**